### DISCUSSION TITLE
Elysia routing not working as expected for HTML5 pushState web apps

### DISCUSSION THREAD

**Discussion by dominictobias** [elysiajs/elysia#757] [Q&A] [answered]
In other route handlers like Express the last route will be executed if other routes didn't match. However in Elysia it is overriding the other routes:

```ts
new Elysia()
  .use(staticPlugin({
    assets: 'public',
    prefix: ''
  }))
  .get('/api', () => 'pong')
  .get('/*', () => Bun.file('public/index.html'))
  .listen(3000)
```

What I expected this to do was:
- if file exists serve that
- if route is /api/... handle those
- finally if nothing else matched redirect to index.html (pushState routing)

What happened:
- All requests return index.html

How can I achieve that with Elysia?

**Comment by crishoj** [comment, marked as answer]
As far as I am aware, Elysia does not yet have an idiomatic way to specify a "default" or "catch-all" route. 

You could abuse the error handler a bit to achieve the same effect though:

```ts
new Elysia()
  .use(staticPlugin({
    assets: 'public',
    prefix: ''
  }))
  .get('/api', () => 'pong')
  .onError(({code}) => {
    if (code === 'NOT_FOUND') return Bun.file('public/index.html')
  })
  .listen(3000)
```
